src/Entity/BannerSelection/Campaign.php

```diff
@@ -34,8 +34,8 @@ public function __construct(
 		$this->end = $end;
 		$this->displayPercentage = $displayPercentage;
 		$this->rng = $rng;
-		$this->buckets = $additionalBuckets;
-		array_unshift( $this->buckets, $firstBucket );
+		$this->buckets[] = $firstBucket;
+		$this->buckets = array_merge( $this->buckets, $additionalBuckets );
```

Comment:
Could be compressed to 

    $this->buckets = array_merge( [$firstBucket], $additionalBuckets );

But not a hard requirement.